site-src/geps/gep-724.md

```diff
@@ -0,0 +1,614 @@
+* Assigns clear responsibilities to Gateway and Route owners.
+* Consistent with pattern of direct references from Routes to all associated
+  resources.
+* Enables attaching Routes to arbitrary parents, such as custom Gateways, other
+  Routes (to be defined), or Meshes.
+* Prevents accidental exposure of Routes.
+* Easy to understand which Gateways/parents a Route is attached to.
+* Further simplifies path to Route inclusion.
+* Follows pattern of direct reference in one direction with a broader trust
+  reference in the other direction.
+* Aligns with initial user feedback.
+
+### Disadvantages
+
+* Attaching a Route to a named listener with SectionName may be a bit confusing.
+* Does not utilize existing ReferencePolicy mechanism.
+* May be more difficult to understand which Routes are attached to a Gateway.
+* Adding/replacing a Gateway requires changes to Routes.
+
+### Potential Expansion
+In the future, it may be useful to add a selector from Route -> Parent. Although
+this would enable greater flexibility, it also significantly increases
+complexity.
+
+## Alternatives
+
+### 1. ReferencePolicy with Gateways selecting Routes
+
+![ReferencePolicy with Gateways selecting Routes](images/724-alt1.png)
+
+A compelling alternative to this proposal would involve retaining the Route
+selector in Gateway and replacing the trust concept in Routes with
+ReferencePolicy. To represent the same example as above, we'd use a Route
+selector on Gateway, a corresponding label on the HTTPRoute, and a
+ReferencePolicy that allowed it:
+
+```yaml
+kind: Gateway
+metadata:
+  name: xlb
+  namespace: infra
+spec:
+  listeners:
+  - name: foo
+    hostname: foo.com
+    port: 80
+    routes:
+      kind: HTTPRoute
+      selector:
+        gateway: xlb
+      namespaces:
+        from: Selector
+        selector:
+          kubernetes.io/metadata.name: foo
+---
+kind: HTTPRoute
+metadata:
+  name: foo
+  namespace: foo
+  labels:
+    gateway: xlb
+spec:
+  rules:
+  - name: abc
+    matches:
+    - path: /bar
+---
+kind: ReferencePolicy
+metadata:
+  name: infra-gateways
+  namespace: foo
+spec:
+  from:
+  - group: networking.gateway.k8s.io
+    kind: Gateway
+    namespace: infra
+  to:
+  - group: networking.gateway.k8s.io
+    kind: HTTPRoute
+```
+
+#### Advantages
+
+* Consistent use of ReferencePolicy throughout the API.
+* Provides a single way of binding Gateways to Routes.
+
+#### Disadvantages
+
+* Even the simplest cross-namespace reference from Gateway -> Route would
+  require a ReferencePolicy in each target namespace.
+* Existing demos and examples would become significantly more verbose.
+* Does not suppport attaching Routes to arbitrary parents.
+* Does not prevent accidental exposure of Routes.
+* Route owners have limited control in terms of which Gateways their Route is
+  attached to.
+
+### 2. ReferencePolicy with Routes referencing Gateways
+
+![ReferencePolicy with Routes referencing Gateways](images/724-alt2.png)
```

Comment:
Are the images for the alternatives swapped or am I understanding these alternatives wrong?